gradebook.cpp

```diff
@@ -7,9 +7,61 @@ grade_Book::grade_Book(){}
 
 grade_Book::grade_Book(std::vector<std::string> assignment_name_list, std::vector<std::string> type, std::vector<int> score_obtained, std::vector<int> max_score){}
        
-double grade_Book::getIndividualGrade(std::string assignment_name){return 0.0;}
-double grade_Book::getCategoryGrade(std::string assignment_type){return 0.0;}
-double grade_Book::getCourseGrade(int option){return 0.0;}
+double grade_Book::getIndividualGrade(std::string assignment_name){
+    //creates an idex of the passed in assignment name
+    int idx;
+    idx = getAssignmentIndex(assignment_name);
+
+
+    //goes to the score_obtained vector to obtain and return the score of that assignment
+    //score as an percentage
+    //double score = ( (score_obtainted[idx] / max_score[idx]) * 100);
+    double score = score_obtainted[idx];
+
+    //print results
+    //std::cout << score << std::endl; //printed
+    return score;
+
+}
+
+double grade_Book::getCategoryGrade(std::string assignment_type){
+    double scoreTotal, maxTotal;
+    scoreTotal = 0;
+    maxTotal = 0;
+    //goes through the scores, if the index of the type equals the passed in assignment type->
+    //it will add that score of the assignment to the scoreTotal, and the max score of the assignment to maxTotal
+    for(int i = 0; i < score_obtainted.size(); i++){
+        if(type[i] == assignment_type){
+            scoreTotal += score_obtainted[i];
+            maxTotal += max_score[i];
+        }
+    }
+
+    double catGrade = scoreTotal / maxTotal;

```

Comment:
This method does not do as what is required. Please refer to either the course website requirements or the Google Doc which has the requirements listed. This method should:

- Print out all assignment grades of the specified type, along with the assignment name
- Print out the category grade

We should probably meet to discuss these functions again because they are more in depth than previously thought.